contrib/drivers/lvm/lvm.go

```diff
@@ -135,16 +141,74 @@ func (d *Driver) PullVolume(volIdentifier string) (*model.VolumeSpec, error) {
 	}, nil
 }
 
+func (d *Driver) geLvInfos() ([]*LvInfo, error) {
+	var lvList []*LvInfo
+	args := []string{"--noheadings", "--unit=g", "-o", "vg_name,name,size", "--nosuffix"}
+	info, err := d.handler("lvs", args)
+	if err != nil {
+		log.Error("Get volume failed", err)
+		return lvList, err
+	}
+	for _, line := range strings.Split(info, "\n") {
+		if len(line) == 0 {
+			continue
+		}
+		words := strings.Fields(line)
+		size, _ := strconv.ParseInt(words[2], 10, 64)
+		lv := &LvInfo{
+			Name: words[0],
+			Vg:   words[1],
+			Size: size,
+		}
+		lvList = append(lvList, lv)
+	}
+	return lvList, nil
+}
+
+func (d *Driver) volumeExists(id string) bool {
+	lvList, _ := d.geLvInfos()
+	name := volumePrefix + id
+	for _, lv := range lvList {
+		if lv.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func (d *Driver) lvHasSnapshot(lvPath string) bool {
+	args := []string{"--noheading", "-C", "-o", "Attr", lvPath}
+	info, err := d.handler("lvdisplay", args)
+	if err != nil {
+		log.Error("Failed to remove logic volume:", err)
+		return false
+	}
+	info = strings.Trim(info, " ")
+	return info[0] == 'o' || info[0] == 'O'
+}
+
 func (d *Driver) DeleteVolume(opt *pb.DeleteVolumeOpts) error {
+
+	id := opt.GetId()
+	if !d.volumeExists(id) {
+		log.Warningf("Volume(%s) does not exist, nothing to remove", id)
+		return nil
+	}
+
 	lvPath, ok := opt.GetMetadata()["lvPath"]
 	if !ok {
-		err := errors.New("Failed to find logic volume path in volume metadata!")
+		err := errors.New("failed to find logic volume path in volume metadata")
 		log.Error(err)
 		return err
 	}
-	if _, err := d.handler("lvremove", []string{
-		lvPath,
-	}); err != nil {
+
+	if d.lvHasSnapshot(lvPath) {
+		err := fmt.Errorf("unable to delete due to existing snapshot for volume: %s", id)
+		log.Error(err)
+		return err
+	}
+
+	if _, err := d.handler("lvremove", []string{"-f", lvPath}); err != nil {
```

Comment:
Please remove '-f' flag

Reply:
If you remove the '-f' flag the handler will return an error code -5. Due to lvremove command  has an interative step that user must enter "Y" to confirm whether to delete it.